### PR TITLE
Lieutenant Fact

### PR DESCRIPTION
**Pull request by srueg**
Relates to https://github.com/projectsyn/lieutenant-api/pull/52

### CODE REVIEW

File: commodore/cluster.py [hunk 1]
```diff
@@ -20,20 +21,26 @@ def fetch_cluster(cfg, clusterid):
...
+        'tenant': target_parameters['customer']['name'],
     }
-    if 'region' in target_data['cloud']:
-        api_resp['facts']['region'] = target_data['cloud']['region']
+    if 'region' in target_parameters['cloud']:
+        api_resp['facts']['region'] = target_parameters['cloud']['region']
+    for cl in target_classes:
+        match_instance = re.search(r'^global\.lieutenant-instance\.(.*)$', cl)
+        if match_instance:
+            api_resp['facts']['lieutenant-instance'] = match_instance.group(1)
```
[the remark this answers is not in the document]
Comment (simu) [reply commodore/cluster.py:43]: Also the whole regex matching is unnecessarily slow. If you really want the regex here, at least use `lieutenant_pattern = re.compile(...)` ahead of the loop and use `lieutenant_pattern.match(cl)` in the loop.

However this is still twice as slow as just prefix matching and string splitting:

I quickly benchmarked both lifting the regex compile out of the loop and string-prefix matching against the original (the break doesn't make much of a difference as the lieutenant-instance is fairly far down the list):

```
import re, timeit, commodore.helpers

target_classes = commodore.helpers.yaml_load('inventory/targets/cluster.yml')['classes']
print(target_classes)

api_resp = {
        'facts': {}
}

def r():
    for cl in target_classes:
        match_instance = re.search(r'^global\.lieutenant-instance\.(.*)$', cl)
        if match_instance:
            api_resp['facts']['lieutenant-instance'] = match_instance.group(1)

def g():
    lt_search = re.compile(r'^global\.lieutenant-instance\.(.*)$')
    for cl in target_classes:
        match_instance = lt_search.match(cl)
        if match_instance:
            api_resp['facts']['lieutenant-instance'] = match_instance.group(1)
            break

def g2():
    for cl in target_classes:
        if cl.startswith('global.lieutenant-instance.'):
            api_resp['facts']['lieutenant-instance'] = cl.split('.')[2]
            break

print(timeit.timeit(r))
print(timeit.timeit(g))
print(timeit.timeit(g2))
```

```
simon@phoenix:~/work/syn/commodore $ pipenv run python test_find_lt.py 
Loading .env environment variables…
['defaults.argocd', 'defaults.backup-cluster-objects', 'defaults.backup-k8up', 'defaults.cert-manager', 'defaults.crossplane', 'defaults.crossplane-cloudscale', 'defaults.csi-cloudscale', 'defaults.espejo', 'defaults.flux', 'defaults.openshift4-oauth', 'defaults.resource-locker', 'defaults.steward', 'defaults.synsights-metrics', 'defaults.synsights-pushgateway', 'defaults.synsights-signalilo', 'defaults.system-upgrade-controller', 'defaults.vault', 'global.common', 'global.eks', 'global.aws', 'global.aws.eu-central-1', 'global.lieutenant-instance.prod', 't-silent-night-6890.c-green-dew-6386']
8.304253647000223
3.0309053789987956
1.521130863999133
```
(Results in sec for 1 million repetitions, i.e. in microseconds per repetition).

So yes, it's not that big of a deal compared to the overall runtime of a Commodore call, but `re` always triggers warning flags in my brain ;)

[the remark this answers is not in the document]
Comment (srueg) [reply commodore/cluster.py]: Fully agree, thanks for the example

**Comment by srueg** [comment]
Yes, I was thinking the same. I guess once we implement #93 we can rework the local mode as well.

File: commodore/cluster.py [hunk 2]
```diff
@@ -20,20 +20,26 @@ def fetch_cluster(cfg, clusterid):
...
+            'url': target_parameters['cluster']['catalog_url'],
         },
-        'tenant': target_data['customer']['name'],
+        'tenant': target_parameters['customer']['name'],
     }
-    if 'region' in target_data['cloud']:
-        api_resp['facts']['region'] = target_data['cloud']['region']
+    if 'region' in target_parameters['cloud']:
+        api_resp['facts']['region'] = target_parameters['cloud']['region']
+    for cl in target_classes:
```
Comment (corvus-ch) [inline commodore/cluster.py:39]: So this is not an actual fact but gets extracted from the classes set. Why is this the case?

Comment (srueg) [reply commodore/cluster.py:39]: That's only done in `--local` mode, where no actual call to the API is made but the existing local file is used and the according API response is generated.

**Pull request merged**